samcli/lib/utils/hash.py

```diff
@@ -51,13 +51,19 @@ def dir_checksum(directory, followlinks=True):
 
     """
     md5_dir = hashlib.md5()
+    files = list()
     # Walk through given directory and find all directories and files.
     for dirpath, _, filenames in os.walk(directory, followlinks=followlinks):
         # Go through every file in the directory and sub-directory.
         for filepath in [os.path.join(dirpath, filename) for filename in filenames]:
             # Look at filename and contents.
             # Encode file's checksum to be utf-8 and bytes.
-            md5_dir.update(filepath.encode("utf-8"))
-            filepath_checksum = file_checksum(filepath)
-            md5_dir.update(filepath_checksum.encode("utf-8"))
+            files.append(filepath)
+
+    files.sort()
```

Comment:
Great catch! I think the native behavior is different for `ls` on mac and linux, can we add a unit test for this? otherwise this looks good.

Reply:
Thanks, added unit tests for sorting changes and from this PR: #2126